### PR TITLE
fix(ds): Persist shard table

### PR DESCRIPTION
**Pull request by ieQu1**
Fixes <issue-or-jira-number>

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl [hunk 1]
```diff
@@ -314,7 +314,7 @@ ensure_tables() ->
         {rlog_shard, ?SHARD},
         {majority, Majority},
         {type, ordered_set},
-        {storage, ram_copies},
+        {storage, disc_copies},
```
Comment (keynslug) [inline apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl:317]: Nice, that answers my question in https://github.com/emqx/emqx/pull/11990#discussion_r1402064094. 👍🏼

File: apps/emqx/src/emqx_persistent_message_ds_replayer.erl [hunk 2]
```diff
@@ -123,7 +123,7 @@ commit_offset(
 
 -spec poll(emqx_persistent_session_ds:id(), inflight(), pos_integer()) ->
     {emqx_session:replies(), inflight()}.
-poll(SessionId, Inflight0, WindowSize) when WindowSize > 0, WindowSize < 16#7fff ->
+poll(SessionId, Inflight0, WindowSize) when WindowSize > 0, WindowSize =< 16#ffff ->
```
Comment (keynslug) [inline apps/emqx/src/emqx_persistent_message_ds_replayer.erl:126]: Oh, I also addressed it in https://github.com/emqx/emqx/pull/12029/files#diff-3d714666e0c828c1fd3b7fd7884623dcf544f641be03e0b0a46153b3ed5fa277R169.

**Review by keynslug — APPROVED** [review]

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl:317]: Q: is there a reason not to use `rocksdb_copies`?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl (hunk 1 above)
Comment (ieQu1) [reply apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl:317]: It's nice to have a ram cache. I don't expect this table to change often to cause the problems normally associated with disc_copies.

**Pull request merged**